src/main/java/net/javadiscord/javabot/api/routes/user_profile/UserProfileController.java

```diff
@@ -0,0 +1,110 @@
+package net.javadiscord.javabot.api.routes.user_profile;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.User;
+import net.javadiscord.javabot.Bot;
+import net.javadiscord.javabot.api.response.ApiResponseBuilder;
+import net.javadiscord.javabot.api.response.ApiResponses;
+import net.javadiscord.javabot.api.routes.CaffeineCache;
+import net.javadiscord.javabot.api.routes.JDAEntity;
+import net.javadiscord.javabot.api.routes.user_profile.model.HelpAccountData;
+import net.javadiscord.javabot.api.routes.user_profile.model.UserProfileData;
+import net.javadiscord.javabot.systems.help.HelpExperienceService;
+import net.javadiscord.javabot.systems.help.model.HelpAccount;
+import net.javadiscord.javabot.systems.moderation.warn.dao.WarnRepository;
+import net.javadiscord.javabot.systems.qotw.QOTWPointsService;
+import net.javadiscord.javabot.systems.qotw.model.QOTWAccount;
+import net.javadiscord.javabot.systems.user_preferences.UserPreferenceService;
+import net.javadiscord.javabot.systems.user_preferences.model.Preference;
+import net.javadiscord.javabot.systems.user_preferences.model.UserPreference;
+import net.javadiscord.javabot.util.ExceptionLogger;
+import net.javadiscord.javabot.util.Pair;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Handles all GET-Requests on the {guild_id}/{user_id} route.
+ */
+@RestController
+public class UserProfileController extends CaffeineCache<Pair<Long, Long>, UserProfileData> implements JDAEntity {
+
+	/**
+	 * The constructor of this class which initializes the {@link Caffeine} cache.
+	 */
+	public UserProfileController() {
+		super(Caffeine.newBuilder()
+				.expireAfterWrite(10, TimeUnit.MINUTES)
+				.build()
+		);
+	}
+
+	/**
+	 * Serves a single users' profile in a specified guild.
+	 *
+	 * @param guildId The guilds' id.
+	 * @param userId  The users' id.
+	 * @return The {@link ResponseEntity}.
+	 */
+	@GetMapping(
+			value = "{guild_id}/{user_id}",
+			produces = MediaType.APPLICATION_JSON_VALUE
+	)
+	public ResponseEntity<String> getUserProfile(
```

Comment:
Spring Web already uses Jackson and is able to serialize objects with that when using `@RestController`.
If you just return a `UserProfileData`-object (`ResponseEntity<UserProfileData>` or `ResponseEntity<?>` if it may be one of multiple types should work as well), Spring will try to serialize that using Jackson.
With that, `produces = MediaType.APPLICATION_JSON_VALUE` might be obsolete.